nucleus/deploy/model_endpoint.py

```diff
@@ -1,12 +1,41 @@
 import concurrent.futures
 from collections import Counter
-from typing import Dict, Optional, Sequence
+from typing import Any, Dict, Optional, Sequence
 
 TASK_PENDING_STATE = "PENDING"
 TASK_SUCCESS_STATE = "SUCCESS"
 TASK_FAILURE_STATE = "FAILURE"
 
 
+class EndpointRequest:
+    """
+    Represents a single request to either a SyncModelEndpoint or AsyncModelEndpoint.
+    Parameters:
+        url: A url to some file that can be read in to a ModelBundle's predict function. Can be an image, raw text, etc.
+        args: A Dictionary with arguments to a ModelBundle's predict function. If the predict function has signature
+            predict_fn(foo, bar), then the keys in the dictionary should be 'foo' and 'bar'. Values must be native Python
+            objects.
+        return_pickled: Whether the output should be a pickled python object, or directly returned serialized json
+        request_id: A user-specifiable id for requests. Should be unique.
+    """
+
+    def __init__(
+        self,
+        url: Optional[str] = None,
+        args: Optional[Dict] = None,
+        return_pickled: Optional[bool] = True,
+        request_id: Optional[str] = None,
+    ):
+        if url is None and args is None:
```

Comment:
Related to the `Union` comment, think this is happening enough where we'd want a helper function.